File: frontend/src/utils/displayMoney.js
import { BASE_CURRENCY, CURRENCY_LABELS } from "../constants/displayCurrencies.js";

const JOD_FORMAT = new Intl.NumberFormat("en-US", {
  minimumFractionDigits: 0,
  maximumFractionDigits: 2,
});

const APPROX_FORMAT = new Intl.NumberFormat("en-US", {
  minimumFractionDigits: 0,
  maximumFractionDigits: 2,
});

export function formatJodAmount(amount, { locale = "ar" } = {}) {
  const n = Number(amount);
  if (!Number.isFinite(n)) return null;
  const formatted = JOD_FORMAT.format(n);
  const suffix = locale === "en" ? "JOD" : CURRENCY_LABELS.JOD;
  return `${formatted} ${suffix}`;
}

export function formatApproximateCurrency(amountJod, targetCurrency, rate) {
  const n = Number(amountJod);
  const r = Number(rate);
  const code = String(targetCurrency || "").toUpperCase();
  if (!Number.isFinite(n) || !Number.isFinite(r) || r <= 0) return null;
  if (!code || code === BASE_CURRENCY) return null;
  const converted = n * r;
  if (!Number.isFinite(converted)) return null;
  const suffix = CURRENCY_LABELS[code] || code;
  return `${APPROX_FORMAT.format(converted)} ${suffix}`;
}

export function shouldShowApproximate(displayCurrency, rate) {
  const code = String(displayCurrency || "").toUpperCase();
  const r = Number(rate);
  return Boolean(code && code !== BASE_CURRENCY && Number.isFinite(r) && r > 0);
}

File: mobile/orderzhouse_app/lib/core/money/display_money.dart
import 'package:intl/intl.dart';

import 'display_currencies.dart';

final NumberFormat _amountFormat = NumberFormat('#,##0.##', 'en_US');

String? formatJodAmount(num? amount) {
  if (amount == null) return null;
  final n = amount.toDouble();
  if (n.isNaN || n.isInfinite) return null;
  return '${_amountFormat.format(n)} ${kCurrencyLabels[kBaseCurrency]}';
}

String? formatApproximateCurrency(num? amountJod, String? targetCurrency, num? rate) {
  if (amountJod == null || rate == null || targetCurrency == null) return null;
  final n = amountJod.toDouble();
  final r = rate.toDouble();
  final code = targetCurrency.toUpperCase();
  if (n.isNaN || r.isNaN || r <= 0) return null;
  if (code.isEmpty || code == kBaseCurrency) return null;
  final converted = n * r;
  if (converted.isNaN || converted.isInfinite) return null;
  final suffix = kCurrencyLabels[code] ?? code;
  return '${_amountFormat.format(converted)} $suffix';
}

bool shouldShowApproximate(String? displayCurrency, num? rate) {
  final code = (displayCurrency ?? '').toUpperCase();
  final r = rate?.toDouble();
  return code.isNotEmpty &&
      code != kBaseCurrency &&
      r != null &&
      !r.isNaN &&
      r > 0;
}
